core/src/main/scala/io/projectglow/vcf/VCFStreamWriter.scala

```diff
@@ -17,46 +17,43 @@
 package io.projectglow.vcf
 
 import java.io.{Closeable, OutputStream}
+import java.util.{ArrayList => JArrayList, HashSet => JHashSet}
 
 import scala.collection.JavaConverters._
 
 import htsjdk.variant.variantcontext.writer.{Options, VariantContextWriter, VariantContextWriterBuilder}
-import htsjdk.variant.variantcontext.{GenotypeBuilder, VariantContext, VariantContextBuilder}
+import htsjdk.variant.variantcontext.{Genotype, GenotypeBuilder, VariantContext, VariantContextBuilder}
 import htsjdk.variant.vcf.{VCFHeader, VCFHeaderLine}
 
 /**
- * This internal row -> variant context stream writer maintains a header whose sample IDs are set
- * exactly once - priority is given to sample IDs provided in the constructor, with the first
- * written variant context used for sample inference if needed.
+ * This internal row -> variant context stream writer maintains a header that is set exactly once. The sample IDs are
+ * set by [[sampleIdInfoOpt]] if provided, or inferred from the first written row otherwise.
  *
- * Upon setting the header sample IDs, any empty sample IDs are replaced with defaults.
+ * If missing sample IDs were used to set the header, sample IDs in all rows to be written will be replaced with those
+ * from the header.
+ * If all sample IDs are present when setting the header, sample IDs in written rows will not be replaced.
+ * Mixed missing/present sample IDs are not permitted.
  *
- * If default header sample IDs were used, any written variant context's empty sample IDs are also
- * replaced with defaults.
+ * @param stream The stream to write to
+ * @param headerLineSet Header lines used to set the VCF header
+ * @param sampleIdInfoOpt Sample IDs (and if any were missing) if pre-determined, None otherwise
```

Comment:
and if any were missing -> and whether they were injected because the ids were missing